### PR TITLE
feat (status_codes): update HTTP status constants wrt RFC 9110

### LINKED ISSUE #2276: Update HTTP status code constants wrt RFC 9110
Update our HTTP status code constants (in [`falcon/status_codes.py`](https://github.com/falconry/falcon/blob/master/falcon/status_codes.py)) to conform to [RFC 9110](https://datatracker.ietf.org/doc/html/rfc9110).

We should be largely up-to-speed, there is at least one known inaccuracy wrt `413 Content Too Large` (see [RFC 9110, Section 15.5.14](https://datatracker.ietf.org/doc/html/rfc9110#name-413-content-too-large)). We still use the older `Payload Too Large`.

### PR DESCRIPTION
**Pull request by prathik2401**
refactor(response): replace falcon.HTTPPayloadTooLarge with falcon.HTTPContentTooLarge

Closes #2276 

# Summary of Changes

Replace the usage of falcon.HTTPPayloadTooLarge with falcon.HTTPContentTooLarge in the codebase. This change aligns with the recent renaming of the class in the Falcon library. The new class name better reflects the purpose of the error, which is to indicate that the content of the request is too large for the server to process.

# Related Issues

#2276 

# Pull Request Checklist

This is just a reminder about the most common mistakes.  Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!

If an item doesn't apply to your pull request, **check it anyway** to make it apparent that there's nothing to do.

- [x] Applied changes to both WSGI and ASGI code paths and interfaces (where applicable).
- [x] Added **tests** for changed code.
- [x] Prefixed code comments with GitHub nick and an appropriate prefix.
- [x] Coding style is consistent with the rest of the framework.
- [x] Updated **documentation** for changed code.
    - [x] Added docstrings for any new classes, functions, or modules.
    - [x] Updated docstrings for any modifications to existing code.
    - [x] Updated both WSGI and ASGI docs (where applicable).
    - [x] Added references to new classes, functions, or modules to the relevant RST file under `docs/`.
    - [x] Updated all relevant supporting documentation files under `docs/`.
    - [x] A copyright notice is included at the top of any new modules (using your own name or the name of your organization).
    - [x] Changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/usage/restructuredtext/directives.html?highlight=versionadded#directive-versionadded).
- [x] Changes (and possible deprecations) have [towncrier](https://towncrier.readthedocs.io/en/latest/quickstart.html#creating-news-fragments) news fragments under `docs/_newsfragments/`, with the file name format `{issue_number}.{fragment_type}.rst`. (Run `towncrier --draft` to ensure it renders correctly.)

If you have *any* questions to *any* of the points above, just **submit and ask**! This checklist is here to *help* you, not to deter you from contributing!

*PR template inspired by the attrs project.*

### CODE REVIEW

File: falcon/__init__.py
Comment (CaselIT) [inline falcon/__init__.py:366]: @vytas7 do you think we should keep the old class as deprecated?

If so the new one should be a subclass of `HTTPPayloadTooLarge` I guess.

Honestly not sure.

Edit: looking at the changelog, a similar thing was done tor 1.4->2.0. let's use the same approach!

File: docs/changes/2.0.0.rst
Comment (CaselIT) [inline docs/changes/2.0.0.rst:171]: Can you revert, since this an old changelog?

**Review by CaselIT — COMMENTED** [review]
Thanks, left a comment and a question for other maintainers

**Comment by CaselIT** [comment]
Can you add a newsfragment? It could read something similar to the 2.0 changelog line

File: falcon/__init__.py
Comment (vytas7) [reply falcon/__init__.py:366]: I definitely think we ought to keep old classes as deprecated, even if 4.0 is a major release, since we have not given the users any deprecation window.

File: docs/changes/2.0.0.rst
Comment (vytas7) [inline docs/changes/2.0.0.rst:367]: Please never edit the content of old changelogs.

File: falcon/errors.py
Comment (vytas7) [inline falcon/errors.py:955]: We must keep the old name for compatibility. Even though 4.0 is a major release that affords breaking changes, we must allow for a longer deprecation window for the publicly documented stuff.

**Review by vytas7 — CHANGES_REQUESTED** [review]
Thanks for this improvement!
We need to address a couple of issues before merging this (commented inline).

**Review by vytas7 — APPROVED** [review]
I have taken the liberty of improving this myself, it should be more or less ready now.

Thanks once again for contributing @prathik2401.

**Review by vytas7 — APPROVED** [review]

File: falcon/__init__.py
Comment (CaselIT) [inline falcon/__init__.py:533]: not in all

**Review by CaselIT — APPROVED** [review]
not sere if we want to add module `__getattr__` in the init to import `HTTP_REQUEST_ENTITY_TOO_LARGE` on demand and give a warning.

same in status?

**Comment by vytas7** [comment]
> not sere if we want to add module `__getattr__` in the init to import `HTTP_REQUEST_ENTITY_TOO_LARGE` on demand and give a warning.
> 
> same in status? (maybe the warning could just be in status module `__getattr__`, the init one just imports it)

No, I think we're good wrt `HTTP_REQUEST_ENTITY_TOO_LARGE` since Falcon 4.0 is a major release, and this constant belongs to an even older iteration, it should have been removed long ago. The previous one is `HTTP_PAYLOAD_TOO_LARGE`.

File: falcon/__init__.py
Comment (vytas7) [reply falcon/__init__.py:533]: Doh, that's not good. I'll address it in the alpha1 PR.

**Pull request merged**
